src/connector/src/parser/mod.rs

```diff
@@ -836,8 +836,26 @@ async fn into_chunk_stream_inner<P: ByteStreamSourceParser>(
                     }
                 },
 
-                Ok(ParseResult::SchemaChange(_)) => {
-                    // TODO
+                Ok(ParseResult::SchemaChange(schema_change)) => {
+                    if schema_change.is_empty() {
+                        continue;
+                    }
+
+                    let (oneshot_tx, oneshot_rx) = tokio::sync::oneshot::channel();
+                    // we bubble up the schema change event to the source executor via channel,
+                    // and wait for the source executor to finish the schema change process before
+                    // parsing the following messages.
+                    if let Some(ref tx) = parser.source_ctx().schema_change_tx {
+                        tx.send((schema_change, oneshot_tx))
+                            .await
+                            .expect("send schema change to executor");
+                        match oneshot_rx.await {
+                            Ok(()) => {}
+                            Err(e) => {
+                                tracing::error!(error = %e.as_report(), "failed to wait for schema change");
```

Comment:
How about adding metrics for auto schema change trigger count, failure count and duration?

Reply:
good point.